src/components/SortCollapse.js

```diff
@@ -3,29 +3,48 @@ import { Button, Col, Collapse, Container, Input, Row, UncontrolledAlert } from
 import PropTypes from 'prop-types';
 
 import { csrfFetch } from 'utils/CSRF';
-import { SORTS } from 'utils/Sort';
+import { SORTS, ORDERED_SORTS } from 'utils/Sort';
 
 import CubeContext from 'contexts/CubeContext';
 import SortContext from 'contexts/SortContext';
 import Query from 'utils/Query';
 
-const SortCollapse = ({ defaultPrimarySort, defaultSecondarySort, defaultSorts, setSorts, ...props }) => {
+const SortCollapse = ({
+  defaultPrimarySort,
+  defaultSecondarySort,
+  defaultTertiarySort,
+  defaultQuaternarySort,
+  defaultSorts,
+  setSorts,
+  ...props
+}) => {
   const [alerts, setAlerts] = useState([]);
   const { canEdit, cubeID } = useContext(CubeContext);
-  const { primary, secondary, changeSort } = useContext(SortContext);
+  const { primary, secondary, tertiary, quaternary, changeSort } = useContext(SortContext);
 
-  const [defSorts, setDefSorts] = useState(
-    defaultSorts && defaultSorts.length > 1 ? [...defaultSorts] : ['Color Category', 'Types-Multicolor'],
-  );
+  const formSorts = (src) => {
+    return [src[0] || 'Color Category', src[1] || 'Types-Multicolor', src[2] || 'CMC2', src[3] || 'Alphabetical'];
+  };
+
+  const [defSorts, setDefSorts] = useState(formSorts(defaultSorts));
 
   const prevSorts = useRef(defSorts);
   useEffect(() => {
     let currentPrimarySort = defaultPrimarySort ?? '';
     let currentSecondarySort = defaultSecondarySort ?? '';
+    let currentTertiarySort = defaultTertiarySort ?? '';
+    let currentQuaternarySort = defaultQuaternarySort ?? '';
     if (!SORTS.includes(currentPrimarySort)) currentPrimarySort = '';
     if (!SORTS.includes(currentSecondarySort)) currentSecondarySort = '';
+    if (!SORTS.includes(currentTertiarySort)) currentTertiarySort = '';
+    if (!ORDERED_SORTS.includes(currentQuaternarySort)) currentQuaternarySort = '';
 
-    if (prevSorts[0] !== currentPrimarySort || prevSorts[1] !== currentSecondarySort) {
+    if (
+      prevSorts[0] !== currentPrimarySort ||
+      prevSorts[1] !== currentSecondarySort ||
+      prevSorts[2] !== currentTertiarySort ||
+      prevSorts[3] !== currentQuaternarySort
+    ) {
```

Comment:
This whole if chain (and all the `Query` calls below) looks like it could be replaced by some `useQueryParam` hooks